doc/release/upcoming_changes/22539.deprecation.rst

```diff
@@ -0,0 +1,24 @@
+``np.find_common_type`` is deprecated
+-------------------------------------
+`numpy.find_common_type` is now deprecated and its use should be replaced
+with either `numpy.result_type` or `numpy.promote_types`.
+Most users leave the second ``scalar_types`` argument to ``find_common_type``
+as ``[]`` in which case ``np.result_type`` and ``np.promote_types`` are both
+faster and more robust.
+When not using ``scalar_types`` the only difference is that the replacement
+intentionally converts non-native byte-order to native byte order.
+
+When the second argument is used things are more complicated.
```

Comment:
Just a minor wording suggestion to better tie it to the case mentioned in the previous paragraph. Feel free to ignore!

```suggestion
When the ``scalar_types`` argument is not ``[]`` things are more complicated.
```